### ISSUE TITLE
Performance degraded after validation was added

### ISSUE THREAD

**Issue by tordans** [openstreetmap/iD#6054]
This is probably similar to https://github.com/openstreetmap/iD/issues/5971.

I noticed that round about the time when validation where added, the editing in Berlin is really slow. This is for both the public version as well as http://preview.ideditor.com/master/#background=Berlin-2018&disable_features=boundaries&map=20.45/52.48086/13.44218.

How can we help figure out what is going on? Would one of those Google Chrome rending-graphs help? I don't know much about them, so I would need a tutorial what to provide.

My system did not change recently, so it's either updates to the OS, Chrome or iD that change the performance.

MacBook Pro (Retina, Mitte 2012), 2,3 GHz Intel Core i7, 16 GB 1600 MHz DDR3
Operating System Mac OSX 10.14.x
Web Browser Chrome 71.0.3578.98

**Update**

- Its really bad when I type my commit message. I type and nothing happens and then the text appears all at once.
- Would screen-videos help?

**Comment by bhousel** [comment]
Yeah this is #5971 - I don't think it should have been closed.

> How can we help figure out what is going on? Would one of those Google Chrome rending-graphs help? I don't know much about them, so I would need a tutorial what to provide.

Yes, profiling in the Google Chrome dev tools would be very helpful.

**Comment by quincylvania** [comment]
From my own testing on this, the degraded performance is almost entirely due to the crossing ways check. iD currently revalidates everything you've edited after every further change, so if you're editing long roads or such this can add up fairly quickly. We have #5901 to only revalidate features you've actually edited since the last validation. We should do that soon.

> Yeah this is #5971 - I don't think it should have been closed.

Oops.

**Comment by bhousel** [comment]
#4731 might be related

**Comment by bhousel** [comment]
This is much better now after #6140 ..

**Comment by tordans** [comment]
@bhousel I used the preview system some more and first thought its just the changeset comment field (the separate issue I created) but now I feel big performance issues in other places.

I tried to debug it better this time.

Location: http://preview.ideditor.com/master/#background=GEOSN-DOP-RGB&disable_features=boundaries&id=w151467550&map=19.97/50.99388/12.80095

**Video:** https://www.dropbox.com/s/azpnxtrdlqly35u/id-performance-reflow.mov?dl=0
There are quite a few seconds between clicking something and being able to use it.

To me it looks like one issue is query?

First edit (outside the video):
<img width="1512" alt="Bildschirmfoto 2019-04-28 um 08 02 27" src="https://user-images.githubusercontent.com/111561/56859605-ab7e5800-698d-11e9-8344-8661b325e31e.png">

Details of the video and after:

<img width="1405" alt="Bildschirmfoto 2019-04-28 um 08 08 31" src="https://user-images.githubusercontent.com/111561/56859612-cb158080-698d-11e9-826e-2f9bb38a36ef.png">

<img width="1403" alt="Bildschirmfoto 2019-04-28 um 08 09 22" src="https://user-images.githubusercontent.com/111561/56859617-d5d01580-698d-11e9-82dd-5a0779ddf367.png">

<img width="1402" alt="Bildschirmfoto 2019-04-28 um 08 09 39" src="https://user-images.githubusercontent.com/111561/56859620-df597d80-698d-11e9-8c2a-f25f4a412074.png">
